Sources/ForageSDK/Foundation/Network/Collector.swift

```diff
@@ -49,7 +55,7 @@ class VGSCollectWrapper: VaultCollector {
         vgsCollect.customHeaders = mutableHeaders
     }
     
-    func sendData(path: String, extraData: [String: Any], completion: @escaping (VaultResponse) -> Void) {
+    func sendData(path: String, vaultAction: VaultAction, extraData: [String: Any], completion: @escaping (VaultResponse) -> Void) {
```

Comment:
> 💡  NOTE: This comment references a comment that I made on `Metrics.swift`.

It seems like there may be an opportunity to hide a lot of the implementation details about how we collect start/end timestamps and even the details that we collect in the trace/log by passing a class instance of `VaultProxyResponseTracer` to `sendData` instead of passing a details we want to capture like `vaultAction`. For illustrative purposes, what do you think of something like:

```swift
func sendData(path: String, extraData: [String: Any], completion: @escaping (VaultResponse) -> Void, tracer: VaultProxyResponseTracer) {
    // code ...

    tracer.start() // calls DispatchTime.now() internally
    vgsCollect.sendData(path: path, extraData: mutableExtraData) { (response) in
        tracer.end() // calls DispatchTime.now() internally
        tracer.setPath(path)
        tracer.setMethod(HttpMethod.post)

        switch response {
        case .success(let code, let data, let urlResponse):
            // code ...
            tracer.setHttpStatusCode(code)
            tracer.captureAsLog()
        case .failure(let code, let data, let urlResponse, let error):
            // code ...
            tracer.setHttpStatusCode(code)
            tracer.captureAsLog()
        }
    }
}
```

Reply:
53be95834a2a44ea544bf379e72075d1138a3f1d